### PR TITLE
Add documentation for RL

### LINKED ISSUE #224: Detailed instructions to run RL

### PR DESCRIPTION
**Pull request by NoB0**
Add information for training dialogue policy via reinforcement learning.

Fixes #224

### CODE REVIEW

File: reinforcement_learning/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  * Gymnasium environments
+  * Training scripts
+
+## Gymnasium environments
+
+To use the environments, you first need to register them with Gymnasium, to do so you need to execute their attached script.
+For example, to register the `DialogueEnvMovieBot-v0` environment, you need to execute the following command:
+
+```bash
+python -m reinforcement_learning.environment.dialogue_env_moviebot
```
Comment (IKostric) [inline reinforcement_learning/README.md:14]: Seems the `gym.register` calls were removed from the mentioned file(s).

File: reinforcement_learning/README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+Currently, there are two reinforcement learning algorithms supported:
+
+  * Deep Q-Learning
+  * Advantage Actor-Critic
+
+Each algorithm has its own trainer, which can be found in the `reinforcement_learning.training` package.
+
+Before training a dialogue policy, you need to prepare a configuration file.
+A default configuration for each supported algorithm is provided in the `config/rl` folder.
```
Comment (IKostric) [inline reinforcement_learning/README.md:33]: Suggestion:
Move the rl specific config to `reinforcement_learning/config`. This way the package is self-containd. global `config` folder contains config for running MovieBot.

File: reinforcement_learning/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+A dialogue policy is trained by executing the following command:
+
+```bash
+python -m reinforcement_learning.train_dialogue_policy -c <training_configuration>
+```
+
+The training logs are stored in Weight and Biases.
```
Comment (IKostric) [inline reinforcement_learning/README.md:41]: Maybe mention that users need to configure W&B to use this option.

File: reinforcement_learning/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+debug: False
+```
+
+**Participants' configuration**: The participants' configuration files are used to specify the parameters of the user simulator and the agent.
+
+**Model's parameters**: The model's parameters are used to specify the parameters of the dialogue policy model. The `algorithm` parameter is used to specify the type of the algorithm used, which can be either `dqn` or `a2c`. The `policy_input_size` parameter is used to specify the size of the input of the policy network. The `hidden_size` parameter is used to specify the size of the hidden layers of the policy network. The `model_path` parameter is used to specify the path to the saved model.
+
+**Hyperparameters**: This section is reserved for training hyperparameters such as learning_rate, discount factor, etc. The hyperparameters are algorithm-specific and can be loaded easily in a dictionary.
+
+**use_intents**: Specifies whether the markovian state should include intents.
```
Comment (IKostric) [inline reinforcement_learning/README.md:77]: User, agent or both? Current or previous turn?

File: reinforcement_learning/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+The user simulator is built on top of [UserSimCRS](https://github.com/iai-group/UserSimCRS/tree/main).
+It is an agenda-based user simulator adapted for reinforcement learning.
+In our case, the user simulator needs to send its response directly to the agent instead of letting the dialogue connector handle it.
+
+We note that its architecture is the same as the one used in the UserSimCRS.
+
+## Try dialogue policy
+
+A dialogue policy can be tested with a human user via the terminal. Instead of interacting with the user simulator, the dialogue policy will interact with the user.
```
Comment (IKostric) [inline reinforcement_learning/README.md:93]: How to configure this? How to run it?

**Review by IKostric — CHANGES_REQUESTED** [review]
Please see comments.

File: reinforcement_learning/README.md (hunk 1 above)
Comment (NoB0) [reply reinforcement_learning/README.md:14]: Good point. I removed this part and updated it with a short description of the available environment.

File: reinforcement_learning/README.md (hunk 2 above)
Comment (NoB0) [reply reinforcement_learning/README.md:33]: Done.

File: reinforcement_learning/README.md (hunk 3 above)
Comment (NoB0) [reply reinforcement_learning/README.md:41]: Done, I also added a link to W&B documentation if needed.

File: reinforcement_learning/README.md (hunk 4 above)
Comment (NoB0) [reply reinforcement_learning/README.md:77]: I added details to the description, please check that it is clarified.

File: reinforcement_learning/README.md (hunk 5 above)
Comment (NoB0) [reply reinforcement_learning/README.md:93]: For now, we do not use a configuration file but it may change.

**Review by IKostric — APPROVED** [review]
LGTM
